source/manual/howto-ssh-to-machines-in-aws.html.md

```diff
@@ -4,81 +4,136 @@ title: SSH into AWS machines
+![Visualisation of commands from dev to production machine](images/govuk-cli-ssh-overview.png)
+
+In AWS, there are no static hostnames, so we can't have `backend-1.backend.integration`
+to SSH to like in Carrenza. EC2 instances have dynamically assigned IPs, which means
+hostnames like `ip-10-1-5-53.eu-west-1.compute.internal`. Each Puppeted instance has a
+"node class" (backend, frontend, ...) and the list of instances belonging to these
 classes is accessible via `govuk_node_list` when logged onto the environment.
 
-To help connecting to the environments, there is a wrapper tool called [`govukcli`](https://github.com/alphagov/govuk-aws/blob/master/tools/govukcli).
+To help connecting to the environments, there is a wrapper tool called
+[`govukcli`](https://github.com/alphagov/govuk-aws/blob/master/tools/govukcli). The tool
+uses the idea of "contexts", where a "context" is a specific environment. By setting a
+context, any subsequent `govukcli` commands will be tied to the environment you chose.
 
-## Install
+## Setup
 
-Ensure you have cloned the `govuk-aws` repository:
+Clone the `govuk-aws` repository and add a symlink to make `govukcli` executable globally:
 
-```
+```sh
 cd ~/govuk
 git clone https://github.com/alphagov/govuk-aws
-```
-
-Add a symlink to ensure you use the current version:
-
-```
 ln -s ~/govuk/govuk-aws/tools/govukcli /usr/local/bin/govukcli
 ```
 
-## Connecting to instances
+You will also need to have followed the [Get SSH access to integration](/manual/get-ssh-access.html)
+instructions.
 
-The tool uses the idea of "contexts", where a "context" is a specific environment.
+## Usage
 
-To view the contexts run:
+### Local dev machine
+
+To view all possible contexts, run:
 
 `govukcli list-contexts`
 
-Set a context to interact with that environment:
+To set a persistent context:
 
 `govukcli set-context integration`
 
-### AWS
+Now when you SSH into an instance, it will be to one running on the specified environment:
 
-In AWS you must specify a node class only. You must use `_` rather than `-` when specifying
-the class.
+```sh
+# SSH into a 'calculators_frontend' node on Integration, as per the context you set
+govukcli ssh calculators_frontend
+```
 
-`govukcli ssh calculators_frontend`
+Alternatively, you can set a context for the current command only by passing a CLI parameter:
 
-This will take you to a random instance of that class.
+```sh
+# SSH into a 'calculators_frontend' node on Staging, ignoing the context you set earlier
+govukcli ssh --context staging calculators_frontend
+```
 
-If you wish to go to a specific node, you can log into the jumpbox:
+Note that the SSH examples above will SSH you into a random machine of the right node class.
+You can find out which class you need by finding the corresponding [app page](https://docs.publishing.service.gov.uk/apps.html).
+`govukcli` finds a matching machine via the 'jumpbox', which you can also SSH into directly:
```

Comment:
I'd consider adding a line break here so that it's easier to read which part of the paragraph the command below refers to.

Reply:
I think the information will flow better if this line about the jumpbox is moved next (or into) the Jumpbox section.